### PR TITLE
fix: fix `byName` resolution

### PR DESCRIPTION
**Pull request by brentleyjones**
In the case of `accessibilitysnapshot`, it depends on `SnapshotTesting` by name. `SnapshotTesting` has an identity of `swift-snapshot-testing` (based on the git url, but it could be different for mirrors/forks), but the name is `snapshottesting`.

### CODE REVIEW

File: swiftpkg/internal/pkginfo_target_deps.bzl [hunk 1]
```diff
@@ -81,15 +81,15 @@ def _resolve_by_name(pkg_ctx, name):
...
+        lambda d: d.name == normalized_name,
     )
     if ext_dep != None:
         return [bazel_labels.new(
             name,
             repository_name = bazel_repo_names.from_identity(ext_dep.identity),
             package = "",
         )]
-    fail("Unable to resolve byName reference {name} in {repo_name}.".format(
+    fail("Unable to resolve byName reference {name} in {repo_name}. {deps}".format(
```
Comment (cgrindel) [inline swiftpkg/internal/pkginfo_target_deps.bzl:92]: Do we need to add the `deps` to the format arguments?

Comment (brentleyjones) [reply swiftpkg/internal/pkginfo_target_deps.bzl:92]: No, this was debug only, sorry, will remove.

**Comment by brentleyjones** [comment]
That's exactly what this change is doing, matching on package name. It was previously trying to match on package identity, which is based on the URL of the package.

File: swiftpkg/internal/pkginfo_target_deps.bzl (hunk 1 above)
Comment (brentleyjones) [inline swiftpkg/internal/pkginfo_target_deps.bzl:92]:
```suggestion
    fail("Unable to resolve byName reference {name} in {repo_name}.".format(
```

**Review by cgrindel — APPROVED** [review]

**Review by luispadron — APPROVED** [review]

**Comment by brentleyjones** [comment]
@cgrindel after this is merged can we get a 0.34.1 release?

**Pull request merged**